### ISSUE TITLE
THT 1.4 - PHP 7 compatibility and Security patches

### ISSUE THREAD

**Issue by xaoseric** [TheHostingTool/TheHostingTool#149]
While I work on the THH 2.0 core, I will also be working on making THT 1.4 compatible with PHP 7, and patching some security holes which will help those that are still using the 1.x series until 2.0 is in a useable state.

**Comment by ponasromas** [comment]
Just fork this project and upgrade it.

**Comment by kmark** [comment]
I appreciate the PR and effort here, especially in regards to security fixes. I'll merge anything you submit that makes sense. Not to discourage your work but I'm wondering if there is enough interest in the current iteration of THT to warrant all this effort on such a legacy codebase. Especially given the ultimate goal is to replace it entirely with THT2.

**Comment by xaoseric** [comment]
While the ultimate goal is to replace it with THT2, it does make sense from a project standpoint to fix issues with the legacy code base in the mean time instead of neglecting it completely. The fixes in the pull request fix the installer, and a few other bugs with there being legacy mysql code so that it is in a working state.

**Comment by kmark** [comment]
Agreed. If you're on the lookout for problem areas I'd look/Google for documented THT exploits and see if they still apply. Unfortunately we've had a few SQL injection attacks in the codebase and I wouldn't be surprised if there were quite a few other vulnerabilities still here.

**Comment by ponasromas** [comment]
@kmark, it is possible to rewrite SQL queries on standardized libraries like MeekroDB in order to mitigate SQL vulnerability. Other security questions should be addressed after basic upgrade to PHP 7. Of course such modifications would require tremendous amount of time and patience to dive in a spaghetti plate :)

**Comment by kmark** [comment]
```
$ rg 'db\->query' | wc -l
     333
```
That's a lot of SQL to go through. Enough for it to be a considerable effort either way. Which route to take is ultimately up to whoever wants to do this. A quick Google brings up [this](https://packetstormsecurity.com/files/134266/TheHostingTool-1.2.6-SQL-Injection.html). They claim they contacted me (and that I didn't reply) but I have no record of receiving anything from them. I'm guessing they sent the notice to some unused email address? Oh well.
